src/routes/api/project/validators.ts

```diff
@@ -0,0 +1,37 @@
+import { Handler } from 'express'
+import BaseValidator from 'validators/baseValidator'
+import * as Joi from '@hapi/joi'
+
+export default class ProjectValidator extends BaseValidator {
+    POST: Handler
+
+    constructor() {
+        super()
+        this.POST = this.requestValidator(this.POSTSchema)
+    }
+
+    POSTSchema = Joi.object({
+        problem: Joi
+            .string()
+            .uri()
+            .required(),
+        submission: Joi
+            .string()
+            .uri()
+            .required(),
+        submissionDirs: Joi
+            .array()
```

Comment:
submissionDirs is just a string and NOT an array. It is a glob string and hence named in plural. I get the confusion.